core/unit_test/TestGraph.hpp

```diff
@@ -813,18 +831,36 @@ struct GraphNodeTypes {
   // Type of a then node.
   using then_t =
       Kokkos::Impl::GraphNodeThenImpl<Exec, ThenFunctor<Kokkos::View<int>>>;
+
+  // Type of a capture node.
+  using capture_t =
+      Kokkos::Impl::GraphNodeCaptureImpl<Exec, CountTestFunctor<Exec>>;
 };
 
+template <typename Exec>
 constexpr bool test_is_graph_kernel() {
-  using types = GraphNodeTypes<TEST_EXECSPACE>;
-  static_assert(Kokkos::Impl::is_graph_kernel_v<types::kernel_t>);
-  static_assert(!Kokkos::Impl::is_graph_kernel_v<types::aggregate_t>);
-  static_assert(Kokkos::Impl::is_graph_kernel_v<types::then_t>,
+  using types = GraphNodeTypes<Exec>;
+  static_assert(Kokkos::Impl::is_graph_kernel_v<typename types::kernel_t>);
+  static_assert(!Kokkos::Impl::is_graph_kernel_v<typename types::aggregate_t>);
+  static_assert(Kokkos::Impl::is_graph_kernel_v<typename types::then_t>,
                 "This should be verified until the 'then' has its own path to "
                 "the driver.");
+  if constexpr (types::support_capture)
+    static_assert(!Kokkos::Impl::is_graph_kernel_v<typename types::capture_t>);
+  return true;
+}
+static_assert(test_is_graph_kernel<TEST_EXECSPACE>());
```

Comment:
Note for the reviewers.

These changes are needed because when capture neither `Cuda`, `HIP` or `SYCL` is defined, there is no definition of `GraphNodeCaptureImpl`.

Since `is_graph_kernel_v` ultimately uses `std::is_base_of_v`, I was getting the error:
>  implicit instantiation of undefined template 'GraphNodeCaptureImpl'

when evaluating `is_graph_kernel_v<typename types::capture_t>`. This is because [of the following rule](https://en.cppreference.com/w/cpp/language/if#Constexpr_if):
> Outside a template, a discarded statement is fully checked.

So I had to make `test_is_graph_kernel` a templated function.